deeplearning4j-zoo/src/main/java/org/deeplearning4j/zoo/model/helper/NASNetHelper.java

```diff
@@ -0,0 +1,192 @@
+import org.deeplearning4j.nn.conf.ConvolutionMode;
+import org.deeplearning4j.nn.conf.graph.ElementWiseVertex;
+import org.deeplearning4j.nn.conf.graph.MergeVertex;
+import org.deeplearning4j.nn.conf.inputs.InputType;
+import org.deeplearning4j.nn.conf.layers.*;
+import org.deeplearning4j.nn.conf.layers.convolutional.Cropping2D;
+import org.deeplearning4j.zoo.model.NASNet;
+import org.nd4j.linalg.activations.Activation;
+import org.nd4j.linalg.primitives.Pair;
+
+import java.util.Map;
+
+/**
+ * Layer helpers {@link NASNet}.
+ *
+ * @author Justin Long (crockpotveggies)
+ */
+public class NASNetHelper {
+
+
+    public static String sepConvBlock(ComputationGraphConfiguration.GraphBuilder graphBuilder, int filters, int kernelSize, int stride, String blockId, String input) {
+        String prefix = "sepConvBlock"+blockId;
+
+        graphBuilder
+                .addLayer(prefix+"_act", new ActivationLayer(Activation.RELU), input)
+                .addLayer(prefix+"_sepconv1", new SeparableConvolution2D.Builder(kernelSize, kernelSize).stride(stride, stride).nOut(filters).hasBias(false)
+                        .convolutionMode(ConvolutionMode.Same).build(), prefix+"_act")
+                .addLayer(prefix+"_conv1_bn", new BatchNormalization.Builder().eps(1e-3).build(), prefix+"_sepconv1")
+                .addLayer(prefix+"_act2", new ActivationLayer(Activation.RELU), prefix+"_conv1_bn")
+                .addLayer(prefix+"_sepconv2", new SeparableConvolution2D.Builder(kernelSize, kernelSize).stride(stride, stride).nOut(filters).hasBias(false)
+                        .convolutionMode(ConvolutionMode.Same).build(), prefix+"_act2")
+                .addLayer(prefix+"_conv2_bn", new BatchNormalization.Builder().eps(1e-3).build(), prefix+"_sepconv2");
+
+        return prefix+"_conv2_bn";
+    }
+
+    public static String adjustBlock(ComputationGraphConfiguration.GraphBuilder graphBuilder, int filters, String blockId, String input) {
+        return adjustBlock(graphBuilder, filters, blockId, input, null);
+    }
+
+    public static String adjustBlock(ComputationGraphConfiguration.GraphBuilder graphBuilder, int filters, String blockId, String input, String inputToMatch) {
+        String prefix = "adjustBlock"+blockId;
+        String outputName = input;
+
+        if(inputToMatch == null) {
+            inputToMatch = input;
+        }
+        Map<String, InputType> layerActivationTypes = graphBuilder.getLayerActivationTypes();
+        int[] shapeToMatch = layerActivationTypes.get(inputToMatch).getShape();
+        int[] inputShape = layerActivationTypes.get(input).getShape();
+
+        if(shapeToMatch[1] != inputShape[1]) {
+            graphBuilder
+                    .addLayer(prefix+"_relu1", new ActivationLayer(Activation.RELU), input)
+                    // tower 1
+                    .addLayer(prefix+"_avgpool1", new SubsamplingLayer.Builder(PoolingType.AVG).kernelSize(2,2).stride(2,2)
```

Comment:
Is this kernel(1,1) in the original? (Not that 1,1 makes sense) - https://github.com/keras-team/keras/blob/master/keras/applications/nasnet.py#L511
Also I'd make truncate mode explicit.